### ISSUE TITLE
Right click on Guard - Crash

### ISSUE THREAD

**Issue by bob0bob** [ldtteam/minecolonies#1761]
minecolonies-universal-
minecraft 1.12.2-
mine colonies 0.8.5810

Crash right clicking on citizen to get their gui.   Tried different citizen and every citizen right click caused crash.

[crash-2017-12-04_21.52.30-client.txt](https://github.com/Minecolonies/minecolonies/files/1529636/crash-2017-12-04_21.52.30-client.txt)
[crash-2017-12-04_21.50.38-client.txt](https://github.com/Minecolonies/minecolonies/files/1529637/crash-2017-12-04_21.50.38-client.txt)

**Comment by bob0bob** [comment]
More info.   It appears to be located to all the citizens assigned to the barracks. 

They also, lost ALL their inventory, including armor.   

You can right click on the hut block and see things there and use the inventory.   They do not get anything from the hut block.

**Comment by bob0bob** [comment]
If I move back the older alpha copy,  everything is fine.  Move back to my save data.  I always backup my data before installing a new alpha version.    Then if I upgrade again to this version.  Something.  Many citizens lose their inventory.  The guards in the barracks gets stuck.  Also 1762 issue appears to be related to this or this build.

**Comment by Raycoms** [comment]
Will take care of this.

**Comment by Raycoms** [comment]
Merged some more fixes.

**Comment by bob0bob** [comment]
I loaded mod minecolonies-universal-1.12.2-0.8.5875.    The right click works now.   It does not crash anymore.

Still doing a test,  the lose of armor and weapons still occured with ALL by barrack guys.  Going to do more testing to 100% make sure.

**Comment by Raycoms** [comment]
Will work on it tomorrow

2017-12-06 0:57 GMT+00:00 bob0bob <notifications@github.com>:

**Comment by bob0bob** [comment]
Ok.   So I tried the following.   loaded my mine colonies v.0.8.5290  .  Took all the armor and weapons from my barrack guys.   Then installed vs. 0.8.5875  and then loaded my 5290 world.   Then I put the items back in their inventory and it worked.    THey took the armor and weapons and started wearing them and patrolling.

But the barracks guard did lose their armor and weapons when I went from 5290 to 5875.
Right click works without crashing now.
